src/compiler/checker.ts

```diff
@@ -7809,7 +7812,8 @@ namespace ts {
                     }
                 }
             }
-            if (!props) {
+            const props = arrayFrom(propSet.values());

```

Comment:
Why not check propSet.size to avoid creating emptyArray to just check size?

Reply:
Sure, why not. Done.